File: system-test/dotnet/Driver.Adapter/External/Erp/Client/ErpRealClient.cs
using Common;
using Driver.Adapter.External.Erp.Client.Dtos;
using Driver.Adapter.External.Erp.Client.Dtos.Error;

namespace Driver.Adapter.External.Erp.Client;

public class ErpRealClient : BaseErpClient
{
    private const string ProductsEndpoint = "/api/products";

    public ErpRealClient(string baseUrl) : base(baseUrl)
    {
    }


    public Task<Result<VoidValue, ExtErpErrorResponse>> CreateProductAsync(ExtCreateProductRequest request)
        => HttpClient.PostAsync(ProductsEndpoint, request);
}

File: system-test/java/src/main/java/com/optivem/shop/testkit/driver/adapter/external/erp/client/ErpRealClient.java
package com.optivem.shop.testkit.driver.adapter.external.erp.client;

import com.optivem.shop.testkit.driver.adapter.external.erp.client.dtos.ExtCreateProductRequest;
import com.optivem.shop.testkit.driver.adapter.external.erp.client.dtos.error.ExtErpErrorResponse;
import com.optivem.shop.testkit.common.Result;

public class ErpRealClient extends BaseErpClient {
    private static final String PRODUCTS_ENDPOINT = "/api/products";

    public ErpRealClient(String baseUrl) {
        super(baseUrl);
    }

    public Result<Void, ExtErpErrorResponse> createProduct(ExtCreateProductRequest request) {
        return httpClient.post(PRODUCTS_ENDPOINT, request);
    }
}
